selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithSpecificBranchTest.java

```diff
@@ -81,59 +64,44 @@ public void setUp() throws Exception {
 
   @AfterClass
   public void deleteTestBranch() throws Exception {
-    if (workspaceServiceClient.exists(gitHubAuxiliaryUserName, testUser.getName())) {
-      testFactoryWithSpecificBranch.delete();
-    }
-  }
-
-  @AfterClass
-  public void restoreContributionTabPreference() throws Exception {
-    testUserPreferencesServiceClient.restoreDefaultContributionTabPreference();
+    testFactoryWithSpecificBranch.delete();
   }
 
   @Test
-  public void factoryWithDirectUrlWithSpecificBranch() throws Exception {
+  public void factoryWithDirectUrlWithSpecificBranch() {
     String repositoryName = testAuxiliaryRepo.getName();
+    final String wsTheiaIdeTerminalTitle = "che-workspace-pod/theia-ide terminal 0";
+
+    testFactoryWithSpecificBranch.authenticateAndOpen();
+
+    theiaIde.switchToIdeFrame();
+    theiaIde.waitTheiaIde();
+    theiaIde.waitLoaderInvisibility();
+    theiaIde.waitNotificationEqualsTo("Che Workspace: Finished cloning projects.");
+    theiaIde.waitNotificationDisappearance(
+        "Che Workspace: Finished cloning projects.", UPDATING_PROJECT_TIMEOUT_SEC);
+
+    theiaProjectTree.waitFilesTab();
+    theiaProjectTree.clickOnFilesTab();
+    theiaProjectTree.waitProjectsRootItem();
+
+    theiaProjectTree.waitItem(repositoryName);
+    theiaProjectTree.openItem(repositoryName);
+    theiaProjectTree.waitItemSelected(repositoryName);
+
+    // check specific branch
+    openTerminalByProposal("che-workspace-pod/theia-ide");
+    theiaTerminal.waitTab(wsTheiaIdeTerminalTitle);
+    theiaTerminal.clickOnTab(wsTheiaIdeTerminalTitle);
+    theiaTerminal.performCommand("cd " + repositoryName);
+    theiaTerminal.performCommand("git status");
+    theiaTerminal.waitTerminalOutput("On branch " + SECOND_BRANCH_NAME, 0);
+  }
 
-    try {
-      testFactoryWithSpecificBranch.authenticateAndOpen();
-    } catch (NoSuchElementException ex) {
-      // remove try-catch block after issue has been resolved
-      fail("https://github.com/eclipse/che/issues/8671");
-    }
-
-    projectExplorer.waitProjectExplorer();
-    notificationsPopupPanel.waitProgressPopupPanelClose();
-    events.clickEventLogBtn();
-
-    events.waitExpectedMessage(
-        "Project " + repositoryName + " imported", UPDATING_PROJECT_TIMEOUT_SEC);
-    events.waitExpectedMessage(
-        "Successfully configured and cloned source code of " + repositoryName,
-        UPDATING_PROJECT_TIMEOUT_SEC);
-    events.waitExpectedMessage(
-        String.format(
-            "Project: %s | cloned from: %s | remote branch: refs/remotes/origin/%s | local branch: %s",
-            repositoryName, repositoryName, SECOND_BRANCH_NAME, SECOND_BRANCH_NAME),
-        UPDATING_PROJECT_TIMEOUT_SEC);
-
-    projectExplorer.waitAndSelectItem(repositoryName);
-    pullRequestPanel.waitOpenPanel();
-
-    projectExplorer.expandPathInProjectExplorer(repositoryName + "/my-lib");
```

Comment:
If there is a reason not to check content of cloned project?

Reply:
Theia doesn't have project types.

Reply:
I see.
What about checking content of repo?

Reply:
I will add content checking to this test.